### PR TITLE
Make kern in MetricsView toggleable with `-`

### PR DESCRIPTION
**Pull request by ctrlcctrlv**
This, in my opinion, solves #3983.

It is still not possible to type a negative integer in LTR order. I
think that making that possible will cause needless disruption to this
file and is also a waste of time. Instead, first type the number, then a
minus sign, which will "toggle" the value negative.

I've documented this as the right way to do it, as well. I understand
why some people might not like this compromise, but it has been broken
for over ten years. If you have a better way to fix it, feel free :-)

### Type of change

- **Bug fix**

### CODE REVIEW

File: fontforgeexe/metricsview.c [hunk 1]
```diff
@@ -1542,16 +1542,26 @@ static int MV_KernChanged(GGadget *g, GEvent *e) {
...
+        int negatives = 0;
+        for (char* p = title; *p != '\0'; p++) {
+            if (*p == '-') negatives++;
+        }
+
+        char* new = str_replace_all(title, "-", "", true); // frees title
+        int val = strtol(new,NULL,10);
+        free(new);
+
+        while (negatives) {
```
Comment (skef) [inline fontforgeexe/metricsview.c]: Wouldn't `if ( negatives%2 )` be preferable here?

File: fontforgeexe/metricsview.c [hunk 2]
```diff
@@ -1542,16 +1542,26 @@ static int MV_KernChanged(GGadget *g, GEvent *e) {
...
-	else {
-	    MV_ChangeKerning(mv,which,val, false);
-	    MVRemetric(mv);
-	}
+        int negatives = 0;
+        for (char* p = title; *p != '\0'; p++) {
+            if (*p == '-') negatives++;
+        }
+
+        char* new = str_replace_all(title, "-", "", true); // frees title
```
Comment (skef) [inline fontforgeexe/metricsview.c]: Unfortunately neither the for loop above nor this call is likely "safe". Looking for a hyphen is looking for a byte value of 45. In UTF 8 that byte could be part of a multibyte character. That will throw off your negative calculation, but more importantly *eliminating* that byte, as with `str_replace_all`, could corrupt the whole string (by shifting one or more characters into the multibyte character).

How much does this matter for a field where someone is supposed to just type a number? Hmm -- maybe not much? Probably not much?

Comment (ctrlcctrlv) [reply fontforgeexe/metricsview.c]: I don't really think this matters. The worst that can happen is a number turns negative, but junk is already being entered. Only ASCII works in this field; as the saying goes, "garbage in, garbage out."

**Comment by skef** [comment]
Consider this re-approved.

**Pull request merged**